src/rez/package_order.py

```diff
@@ -614,13 +633,13 @@ def from_pod(cls, data):
         )
 
 
-class PackageOrderList(list):
+class PackageOrderList(List[PackageOrder]):
```

Comment:
Note that we use `typing.List` here instead of `list` because `list` does not become indexable at runtime until python 3.9.   It's still safe to use `list[X]` inside annotations as long as we use `from __future__ import annotations`.